app/models/warehouse_report/outcomes/base.rb

```diff
@@ -334,8 +339,11 @@ def returns_to_shelter(leaver_scope)
       leavers_with_date = leaver_scope.pluck(:client_id, :housing_exit).to_h
       return {} unless leavers_with_date.present?
 
-      returner_ids = Reporting::Return.where(client_id: leavers_with_date.keys).distinct.pluck(:client_id)
-
+      returner_ids = Reporting::Return.where(client_id: leavers_with_date.keys).
+        distinct.
+        pluck(:client_id)
+      returner_demographics = Reporting::Return.where(client_id: returner_ids).distinct.
+        pluck(:client_id, :race, :ethnicity, :gender).index_by(&:first)
```

Comment:
Add a comment to note the order matters here> The pluck and the use of the array positions are separated a bit here...

Reply:
Good point.